classes/controller/FrontController.php

```diff
@@ -1317,11 +1333,15 @@ public function getLayout()
                 'content_only' => $content_only,
             ]
         )) {
+            // Modules can return data with or without the full path
+            // We will strip it if provided
+            $overridden_layout = str_replace(['layouts/', '.tpl'], '', $overridden_layout);
```

Comment:
Since this hook was initially meant to override the layout path, and not the layout name, maybe this should be extracted from this `getLayoutName` and be handled in the `getLayout` method instead These are two different use cases, the hook was never meant to return a layout name

Reply:
@jolelievre Yes, but we will need to get the layout name anyway in this case, so what should we do?

Either just rely on this method of extracting it from path or making another hook for the layout name only - which is a complication.

I think it will be fine, because there are many places in the core that have hardcoded /layouts folder, the structure is solid and won't be changed I think.